src/interfaces/messages/messages/messages-get.ts

```diff
@@ -0,0 +1,53 @@
+import type { SignatureInput } from '../../../jose/jws/general/types.js';
+import type { MessagesGetDescriptor, MessagesGetMessage } from '../types.js';
+
+import { parseCid } from '../../../utils/cid.js';
+import { validateAuthorizationIntegrity } from '../../../core/auth.js';
+import { DwnInterfaceName, DwnMethodName, Message } from '../../../core/message.js';
+
+export type MessagesGetOptions = {
+  messageCids: string[];
+  authorizationSignatureInput: SignatureInput;
+};
+
+export class MessagesGet extends Message<MessagesGetMessage> {
+  public static async parse(message: MessagesGetMessage): Promise<MessagesGet> {
+    Message.validateJsonSchema(message);
+    this.validateMessageCids(message.descriptor.messageCids);
+
+    await validateAuthorizationIntegrity(message);
+
+    return new MessagesGet(message);
+  }
+
+  public static async create(options: MessagesGetOptions): Promise<MessagesGet> {
+    const descriptor: MessagesGetDescriptor = {
+      interface   : DwnInterfaceName.Messages,
+      method      : DwnMethodName.Get,
+      messageCids : options.messageCids
+    };
+
+    const authorization = await Message.signAsAuthorization(descriptor, options.authorizationSignatureInput);
+    const message = { descriptor, authorization };
+
+    Message.validateJsonSchema(message);
+    this.validateMessageCids(options.messageCids);
+
+    return new MessagesGet(message);
+  }
+
+  /**
+   * Q for reviewers: worth it?
```

Comment:
Eh, might as well. Seems harmless. I might even go so far as to add a custom `format` to the JSON schema.

Reply:
i dont think we do this kind of validation anywhere else (e.g. `dataCid`) so i think it's probably fine to not do it

Reply:
reason behind doing so was to prevent unnecessary i/o for bunk CIDs

Reply:
@dcrousso imo `dataCid` is validated when its integrity checked [here](https://github.com/TBD54566975/dwn-sdk-js/blob/main/src/store/storage-controller.ts#L70-L81).

Reply:
@diehuxx definitely considered adding a custom `format` to JSON schema but decided against it. Trying to keep use of json schema as language agnostic as possible in hopes that they can be used if anyone takes a stab at implementing any part of DWNs in another language